support-services/src/main/scala/com/gu/support/redemption/generator/GiftCodeGenerator.scala

```diff
@@ -27,6 +28,11 @@ object GiftDuration {
 
   case object Gift12Month extends GiftDuration("12")
 
+  def fromBillingPeriod = (billingPeriod: BillingPeriod) => billingPeriod match {
```

Comment:
two points:
in terms of dependencies, at the moment this generator is completely standalone.  The code to map billing periods is more how we want to interface with the generator based on what the workers input is so is not really generation code.  I would prefer to see this function in the subscription builder, as that is the code responsible for making a generated code from whatever input we already have.

Secondly I think this could just be a normal method:
```suggestion
  def fromBillingPeriod(billingPeriod: BillingPeriod): GiftDuration = billingPeriod match {
```